api/v1beta1/solrcollectionalias_types.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2019 Bloomberg Finance LP.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SolrCollectionAliasSpec defines the desired state of SolrCollectionAlias
+type SolrCollectionAliasSpec struct {
+	// A reference to the SolrCloud to create alias on
+	SolrCloud string `json:"solrCloud"`
```

Comment:
We should eventually support clouds not running in kube, via a zookeeper connection string. But we also need to do this for the `SolrCollection` CRD.

This can be a feature added in the future.

Reply:
I see, so in future make this optional and couple with another input that is a zk connection string. Perhaps add in a new issue?

Reply:
I see it as adding a solrCloud field with lets you either specify a ZkConnectionString, URL or NamespacedName.

We can make it backwards compatible for a bit whenever we make the change. But we should make sure it's comprehensive and add it for both the Collection and Alias at that same time. I'll make an issue